Framework/CurveFitting/test/Functions/MeierTest.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/FunctionDomain1D.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/FunctionValues.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidCurveFitting/Functions/Meier.h"
+
+#include <array>
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+using namespace Mantid::CurveFitting;
+using namespace Mantid::CurveFitting::Functions;
+using namespace Mantid::CurveFitting::Algorithms;
+
+class MeierTest : public CxxTest::TestSuite {
+public:
+  void testFunctionName() {
+    auto meier = createFunction();
+    TS_ASSERT_EQUALS(meier->name(), "Meier");
+  }
+
+  void testFunctionCategory() {
+    auto meier = createFunction();
+    TS_ASSERT_EQUALS(meier->category(), "Muon\\MuonSpecific");
+  }
+
+  void testFunctionRegisteredInFactory() { FunctionFactory::Instance().createInitialized("name=Meier"); }
+
+  void testFunctionHasExpectedOrderedParameters() {
+    auto meier = createFunction();
+
+    const size_t expectedNParams = 5;
+    TS_ASSERT_EQUALS(meier->nParams(), expectedNParams);
+
+    if (meier->nParams() == expectedNParams) {
+      const char *expectedParams[expectedNParams] = {"A0", "FreqD", "FreqQ", "Sigma", "Lambda"};
+      auto actualParamsNames = meier->getParameterNames();
+
+      for (size_t i = 0; i < expectedNParams; ++i) {
+        TS_ASSERT_EQUALS(actualParamsNames[i], expectedParams[i]);
+      }
+    }
+  }
+
+  void testFunctionHasExpectedOrderedAttributes() {
+    auto meier = createFunction();
+
+    const size_t expectedNAttrs = 1;
+    TS_ASSERT_EQUALS(meier->nAttributes(), expectedNAttrs);
+
+    if (meier->nAttributes() == expectedNAttrs) {
+      const char *expectedAttrs[expectedNAttrs] = {"Spin"};
+      auto actualAttrsNames = meier->getAttributeNames();
+
+      for (size_t i = 0; i < expectedNAttrs; ++i) {
+        TS_ASSERT_EQUALS(actualAttrsNames[i], expectedAttrs[i]);
+      }
+    }
+  }
+
```

Comment:
can we have a test that gives an invalid attribute value (e.g. 0.6) to make sure its handled gracefully.